index.bs

```diff
@@ -500,13 +502,15 @@ The <dfn method for=Window>getScreenDetails()</dfn> method steps are:
 
 1. Let |promise| be [=/a new promise=].
 
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=policy-controlled feature=] named "[=permission-policy/window-management=]", then [=/reject=] |promise| with a {{"NotAllowedError"}} {{DOMException}} and abort these steps.
+
 1. Run the following steps [=/in parallel=]:
 
     1. Let |permissionState| be [=/request permission to use=] `"window-management"`.
 
-    1. If |permissionState| is "{{PermissionState/denied}}" then [=/reject=] |promise| with a {{"NotAllowedError"}} {{DOMException}} and abort these steps.
+    1. If |permissionState| is "{{PermissionState/denied}}", then [=/queue a global task=] on the [=/relevant global object=] of [=/this=] using the [=/window placement task source=] to [=/reject=] |promise| with a {{"NotAllowedError"}} {{DOMException}} and abort these steps.
```

Comment:
Rather than queuing a task in each step, how about you write something like,

> 1. Let permissionState be request permission to use "window-management".
> 2. Queue a global task on the relevant global object of this using the window placement task source to run the following steps,
>    1. If permissionState is...
>    2. Etc...

The step where you are modifying this.[[screenDetails]] needs to happen back on the "main thread" as well.

Reply:
Done, that seems like a nice bit of refactoring. I actually removed the step about prompt and transient activation, because those details should be encapsulated in [request permission to use](https://w3c.github.io/permissions/#dfn-request-permission-to-use), which explicitly states `This algorithm returns either "granted" or "denied".` (i.e. it doesn't return "prompt").